floris/simulation/wake_deflection/base_velocity_deflection.py

```diff
@@ -189,18 +189,155 @@ def calculate_effective_yaw_angle(
+            HH = turbine.hub_height
+            V = flow_field.v
+            W = flow_field.w
+            Uinf = np.mean(flow_field.wind_map.grid_wind_speed)
+
+            # print('ET:', np.mean(V))
+
+            eps = self.eps_gain * D  # Use set value
+            idx = np.where(
+                (np.abs(x_locations - coord.x1) < 1)
+                & (np.abs(y_locations - coord.x2) <= D / 2)
+                & (np.abs(z_locations - coord.x3) <= D / 2)
+            )
+
+            # yLocs = y_locations[idx] + 0.01 - coord.x2
+
+            # location of left vortex
+            yL = y_locations[idx] + 0.01 - (coord.x2) - (D / 2)
+            zL = z_locations[idx] + 0.01 - HH
+            zLG = z_locations[idx] + 0.01 + HH
+            rL = yL ** 2 + zL ** 2
+            rLG = yL ** 2 + zLG ** 2
+
+            # location of right vortex
+            yR = y_locations[idx] + 0.01 - (coord.x2) + (D / 2)
+            zR = z_locations[idx] + 0.01 - HH
+            zRG = z_locations[idx] + 0.01 + HH
+            rR = yR ** 2 + zR ** 2
+            rRG = yR ** 2 + zRG ** 2
+
+            # wake rotation vortex
+            # zC = z_locations[idx] + 0.01 - (HH)
+            # rC = yLocs ** 2 + zC ** 2
+
+            # find wake deflection from CRV
+            test_gamma = np.linspace(-45, 45, 91)
+            # print(V[idx])
+            avg_V = np.mean(V[idx])
+            avg_W = np.mean(W[idx])
+            target_tilt_ix = None
+
+            # what tilt angle would have produced that same average spanwise velocity
+            tilt = test_gamma
+            minTilt = 100000.0
+
+            for i in range(len(tilt)):
+
+                Gamma_left = (
+                    (np.pi / 8) * D * Uinf * Uinf * Ct * sind(tilt[i]) * cosd(tilt[i])
+                )
+                Gamma_right = (
+                    -(np.pi / 8) * D * Uinf * Uinf * Ct * sind(tilt[i]) * cosd(tilt[i])
+                )
+
+                Weff = (
+                    (-yL * Gamma_left)
+                    / (2 * np.pi * rL)
+                    * (1 - np.exp(-rL / (eps ** 2)))
+                    + (-yR * Gamma_right)
+                    / (2 * np.pi * rR)
+                    * (1 - np.exp(-rR / (eps ** 2)))
+                    + (zLG * -Gamma_left)
+                    / (2 * np.pi * rLG)
+                    * (1 - np.exp(-rLG / (eps ** 2)))
+                    + (zRG * -Gamma_right)
+                    / (2 * np.pi * rRG)
+                    * (1 - np.exp(-rRG / (eps ** 2)))
+                )
+
+                tmp = np.abs(avg_W - 2*np.mean(Weff))
+
+                if tmp < minTilt:
+                    target_tilt_ix = i
+                    minTilt = tmp
+
+            if flow_field.tilt_eff is False:
+                return turbine.tilt_angle
+            elif target_tilt_ix is not None:
+                tilt_effective = test_gamma[target_tilt_ix]
+            else:
+                err_msg = "No effective yaw angle is found. Set to 0."
```

Comment:
Error message should be updated to "tilt angle"

Reply:
yes you're right

Reply:
Think this is now fixed